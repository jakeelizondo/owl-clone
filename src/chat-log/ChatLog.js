import React from 'react';
import Message from '../message/Message.js';
import renderer from 'react-test-renderer';

function ChatLog(props) {
  let testDate = new Date(1548852724247);
  console.log(testDate.getHours());
  console.log(testDate.getMinutes());

  let messagesArray = props.chatEvents.map((chat) => {
    // isolate username
    let chatUser = Object.assign(
      props.participants.filter((person) => person.id === chat.participantId)
    );
    let chatUserName = chatUser[0].name;
    let chatAvatar = chatUser[0].avatar;

    // take timestamp, get hours and minutes, concatenate into string of time
    let chatDate = new Date(chat.timestamp);
    let chatHour = chatDate.getHours();
    let chatMinutes = chatDate.getMinutes();
    let chatTime = `${chatHour}:${chatMinutes}`;
    console.log(chatTime);

    return (
      <Message
        key={chat.id}
        name={chatUserName}
        time={chatTime}
        type={chat.type}
        message={chat.message ? chat.message : null}
        avatar={chatAvatar}
      />
    );
  });

  // console.log(messagesArray);
  return <div className="messages-list">{messagesArray}</div>;
}

export default ChatLog;
