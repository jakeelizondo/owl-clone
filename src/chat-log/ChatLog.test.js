import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import ChatLog from './ChatLog';
import participants from '../participants';
import chatEvents from '../chatEvents';

describe('ChatLog Component', () => {
  // smoke test for app component

  it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(
      <ChatLog chatEvents={chatEvents} participants={participants} />,
      div
    );
    ReactDOM.unmountComponentAtNode(div);
  });

  // snapshot test
  it('renders the UI as expected', () => {
    const tree = renderer
      .create(<ChatLog chatEvents={chatEvents} participants={participants} />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
