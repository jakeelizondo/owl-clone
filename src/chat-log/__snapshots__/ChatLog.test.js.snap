// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ChatLog Component renders the UI as expected 1`] = `
<div
  className="messages-list"
>
  <div
    className="chat-message"
  >
    <img
      className="chat-avatar"
      src="https://robohash.org/utlaborumfugit.jpg?size=200x200&set=set1"
    />
    <p>
      Axe Kubicka
    </p>
    <p>
      7:48
    </p>
    <p>
      Hello world
    </p>
  </div>
  <div>
    <p>
      Caty Flucker
       gave a 
      thumbs-up
    </p>
  </div>
  <div>
    <p>
      Frank Runciman
       gave a 
      thumbs-down
    </p>
  </div>
  <div>
    <p>
      Koren Templeton
       raised their hand
    </p>
  </div>
  <div>
    <p>
      Koren Templeton
       clapped
    </p>
  </div>
  <div>
    <p>
      Koren Templeton
       joined the stage
    </p>
  </div>
  <div>
    <p>
      Ashla Attwool
       left the stage
    </p>
  </div>
  <div>
    <p>
      Axe Kubicka
       joined the stage
    </p>
  </div>
  <div>
    <p>
      Axe Kubicka
       left the stage
    </p>
  </div>
</div>
`;
